src/common/SurgeSynthesizer.cpp

```diff
@@ -3927,14 +3932,32 @@ void loadPatchInBackgroundThread(SurgeSynthesizer *sy)
         }
         else
         {
-            synth->loadPatchByPath(synth->patchid_file, -1, s.c_str());
+            synth->loadPatchByPath(synth->patchid_file, -1, pathstr.c_str());
         }
     }
 
     synth->storage.getPatch().isDirty = false;
     synth->patchChanged = true;
     synth->halt_engine = false;
 
+    // Notify the 'patch loaded' listener(s)
+    if (patchid >= 0)
+    {
+        Patch p = synth->storage.patch_list[synth->patchid];
+        pathstr = path_to_string(p.path);
+        // Trim extension
+        pathstr = pathstr.substr(0, pathstr.find_last_of("."));
```

Comment:
Turns out doing this in a string based way is a bit fragile. It works on our current OSes but you can use `filesystem::path.replace_extension("")` (like this)[https://en.cppreference.com/w/cpp/filesystem/path/replace_extension] before you convert to a string. (Generally, and mostly thanks to windows, string math on paths is hard because of UTF16 windows file system. We utf8ify early with path_to_string)